kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/convert/BigQuerySchemaConverter.java

```diff
@@ -120,33 +119,37 @@ public BigQuerySchemaConverter(boolean allFieldsNullable) {
     return com.google.cloud.bigquery.Schema.of(fields);
   }
 
-  private void throwOnCycle(Schema kafkaConnectSchema, List<Schema> seenSoFar) {
+  private boolean throwOnCycle(Schema kafkaConnectSchema, List<Schema> seenSoFar) {
     if (PRIMITIVE_TYPE_MAP.containsKey(kafkaConnectSchema.type())) {
-      return;
+      return false;
     }
 
     if (seenSoFar.contains(kafkaConnectSchema)) {
-      throw new ConversionConnectException("Kafka Connect schema contains cycle");
+      logger.error("Cycle detected : " + kafkaConnectSchema.name());
+      return true;
     }
 
     seenSoFar.add(kafkaConnectSchema);
+    boolean hasCycle = false;
     switch(kafkaConnectSchema.type()) {
       case ARRAY:
-        throwOnCycle(kafkaConnectSchema.valueSchema(), seenSoFar);
+        hasCycle = throwOnCycle(kafkaConnectSchema.valueSchema(), seenSoFar);
         break;
       case MAP:
-        throwOnCycle(kafkaConnectSchema.keySchema(), seenSoFar);
-        throwOnCycle(kafkaConnectSchema.valueSchema(), seenSoFar);
+        hasCycle = throwOnCycle(kafkaConnectSchema.keySchema(), seenSoFar);
+        hasCycle = hasCycle || throwOnCycle(kafkaConnectSchema.valueSchema(), seenSoFar);
         break;
       case STRUCT:
-        kafkaConnectSchema.fields().forEach(f -> throwOnCycle(f.schema(), seenSoFar));
+        List<Boolean> cyclesBool = kafkaConnectSchema.fields().stream().map(f -> throwOnCycle(f.schema(), seenSoFar)).collect(Collectors.toList());
+        hasCycle = cyclesBool.stream().anyMatch(aBoolean -> aBoolean);
```

Comment:
Nit: no need for the intermediate call to `Stream::collect`:

```suggestion
        hasCycle = kafkaConnectSchema.fields().stream().anyMatch(f -> throwOnCycle(f.schema(), seenSoFar));
```